src/Azure.Functions.Cli/Actions/HostActions/StartHostAction.cs

```diff
@@ -395,25 +408,53 @@ internal static async Task CheckNonOptionalSettings(IEnumerable<KeyValuePair<str
             }
         }
 
-        private async Task<(Uri listenUri, Uri baseUri, X509Certificate2 cert)> Setup()
+        /// <summary>
+        /// Add additional settings to be consumed by the EasyAuth Middleware
+        /// </summary>
+        /// <param name="baseUri"></param>
+        private static void SetupMiddlewareConfig(Uri baseUri)
+        {
+            // These are not necessary when Easy Auth is used in conjunction with the Functions CLI
+            // We add them here for compatibility with existing Easy Auth / Middleware codebase
+            var args = new string[] {
+                string.Format("{0}={1}", Constants.MiddlewareListenUrlSetting, baseUri.ToString()),
+                string.Format("{0}={1}", Constants.MiddlewareHostUrlSetting, baseUri.ToString()),
+            };
+
+            // Add above arguments, as well as the settings from local.middleware.json
+            IConfigurationRoot config = new ConfigurationBuilder()
+                .SetBasePath(Directory.GetCurrentDirectory())
+                .AddJsonFile(SecretsManager.MiddlewareAuthSettingsFilePath)
+                .AddCommandLine(args)
+                .Build();
+
+            ModuleConfig.HostConfig = config;
+        }
+
+        private async Task<(Uri listenUri, Uri baseUri, X509Certificate2 cert, string path, string password)> Setup()
         {
             var protocol = UseHttps ? "https" : "http";
-            X509Certificate2 cert = UseHttps
-                ? await SecurityHelpers.GetOrCreateCertificate(CertPath, CertPassword)
-                : null;
-            return (new Uri($"{protocol}://0.0.0.0:{Port}"), new Uri($"{protocol}://localhost:{Port}"), cert);
+            if (UseHttps)
+            {
+                (X509Certificate2 cert, string certPath, string certPassword) = await SecurityHelpers.GetOrCreateCertificate(CertPath, CertPassword);
```

Comment:
It feels unnatural to fetch the certificate and both its path and password. Is this because the Easy Auth code you're calling expects these? Could the Easy Auth code be refactored to take an `X509Certificate2` for embedded scenarios like this?

Reply:
Removing all of this altogether. Easy Auth, when truly middleware (which it is now), doesn't need its own reference to the certificate.